secure-storage/secure-storage-impl/src/main/java/com/duckduckgo/securestorage/impl/RealSecureStorage.kt

```diff
@@ -25,75 +25,91 @@ import com.duckduckgo.securestorage.impl.encryption.EncryptionHelper.EncryptedSt
+            secureStorageRepository?.addWebsiteLoginCredential(websiteLoginDetailsWithCredentials.toDataEntity())
         }
     }
 
-    override suspend fun websiteLoginDetailsForDomain(domain: String): Flow<List<WebsiteLoginDetails>> =
-        withContext(dispatchers.io()) {
-            secureStorageRepository.websiteLoginCredentialsForDomain(domain).map { list ->
-                list.map {
-                    it.toDetails()
+    override suspend fun websiteLoginDetailsForDomain(domain: String): Flow<List<WebsiteLoginDetails>> {
+        return if (secureStorageRepository != null) {
+            withContext(dispatchers.io()) {
+                secureStorageRepository!!.websiteLoginCredentialsForDomain(domain).map { list ->
+                    list.map {
+                        it.toDetails()
+                    }
                 }
             }
-        }
+        } else emptyFlow()
+    }
 
     override suspend fun websiteLoginDetails(): Flow<List<WebsiteLoginDetails>> =
-        withContext(dispatchers.io()) {
-            secureStorageRepository.websiteLoginCredentials().map { list ->
-                list.map {
-                    it.toDetails()
+        if (secureStorageRepository != null) {
+            withContext(dispatchers.io()) {
+                secureStorageRepository!!.websiteLoginCredentials().map { list ->
+                    list.map {
+                        it.toDetails()
+                    }
                 }
             }
-        }
+        } else emptyFlow()
 
     override suspend fun getWebsiteLoginDetailsWithCredentials(id: Int): WebsiteLoginDetailsWithCredentials? =
         withContext(dispatchers.io()) {
-            secureStorageRepository.getWebsiteLoginCredentialsForId(id)?.toCredentials()
+            secureStorageRepository?.getWebsiteLoginCredentialsForId(id)?.toCredentials()
         }
 
     override suspend fun websiteLoginDetailsWithCredentialsForDomain(domain: String): Flow<List<WebsiteLoginDetailsWithCredentials>> =
-        withContext(dispatchers.io()) {
-            secureStorageRepository.websiteLoginCredentialsForDomain(domain).map { list ->
-                list.map {
-                    it.toCredentials()
+        if (secureStorageRepository != null) {
+            withContext(dispatchers.io()) {
+                secureStorageRepository!!.websiteLoginCredentialsForDomain(domain).map { list ->
+                    list.map {
+                        it.toCredentials()
+                    }
                 }
             }
-        }
+        } else emptyFlow()
 
     override suspend fun websiteLoginDetailsWithCredentials(): Flow<List<WebsiteLoginDetailsWithCredentials>> =
-        withContext(dispatchers.io()) {
-            secureStorageRepository.websiteLoginCredentials().map { list ->
-                list.map {
-                    it.toCredentials()
+        if (secureStorageRepository != null) {
```

Comment:
Could use `secureStorageRepository?.let` instead of the null check then the `!!` usage